types/highcharts/index.d.ts

```diff
@@ -6134,7 +6134,7 @@ declare namespace Highcharts {
         /**
          * The y value of the point.
          */
-        y?: number;
+        y?: number | null;
```

Comment:
Correct, y can be null:

https://github.com/highcharts/highcharts/blob/8f802d8c7/js/parts/Series.js#L5730-#L5736